addons/account/models/account_bank_statement.py

```diff
@@ -422,7 +422,11 @@ def button_reopen(self):
             raise UserError(_("Only validated statements can be reset to new."))
 
         self.write({'state': 'open'})
+        names = self.line_ids.move_id.mapped('name')
         self.line_ids.move_id.button_draft()
+        self.line_ids.move_id.posted_before = False
+        for name, move in zip(names, self.line_ids.move_id):
+            move.name = name
```

Comment:
In my opinion, `posted_before` should stay at `True` but we need another way to write the journal items. I don't understand which error are you facing when reseting the statement to draft.